hoodie-client/src/main/java/com/uber/hoodie/table/HoodieMergeOnReadTable.java

```diff
@@ -527,6 +525,10 @@ private HoodieRollbackStat rollback(HoodieIndex hoodieIndex, String partitionPat
               if (writer != null) {
                 writer.close();
               }
+              if (success) {
```

Comment:
Can you leave comments here detailing the reason we are doing this is due to different file system guarantees of HDfs and cloud ?

Reply:
better yet, can you special class this based on the StorageScheme, instead of handling via an exception>?

Reply:
should this be in a `finally` clause instead?

Reply:
+1 to @n3nash 's point. but I think its generally safe to do this. even with hdfs, length is not updates until the hflush/hsync etc is called..

Reply:
@n3nash @vinothchandar : Added documentation the reason for this change

@vinothchandar : This writer.close() part is in finally() block. I am not sure if we need to update filesToNumBlocksRollback if writer.close() fails as the rollback would fail in that case. If still needed, would be fine with adding.

Reply:
No I don;t think its needed..